unity-renderer/Assets/Scripts/MainScripts/DCL/SignRequest/RPCSignRequest.cs

```diff
@@ -0,0 +1,42 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using System.Threading;
+using Decentraland.Renderer.KernelServices;
+using System;
+using System.Collections.Generic;
+using SignBodyRequest = Decentraland.Renderer.KernelServices.SignBodyRequest;
+using SignBodyResponse = Decentraland.Renderer.KernelServices.SignBodyResponse;
+
+public class RPCSignRequest : IRPCSignRequest
+{
+    private const int REQUEST_TIMEOUT = 30;
+
+    private static RPCSignRequest i;
+
+    private readonly IRPC rpc;
+
+    public static RPCSignRequest CreateSharedInstance(IRPC rpc)
+    {
+        i = new RPCSignRequest(rpc);
+        return i;
```

Comment:
What's the use case for this `CreateSharedInstance`? We are trying to move away from singletons, is it needed?

If needed, shouldnt we check if `i` exists and return it instead?